### PR TITLE
Pmyip3/page to import publications using g scholar ms

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
When the import publication button is clicked and after the profile link is entered(accept any values for now), the hardcoded publications in json file will be displayed, which is pretended to be imported by scholar api (to be implemented).
The user should be able to click on the ones they want to insert and then when they're done, they click a button that sends that data to the backend to do a mass insert into the db. 

# Type of change 
- [ ] Bug fix
- [X] New feature
- [ ] Refactoring
- [ ] Documentation 

# Problem
We need the feature of importing publications as per the client request. 

# Solution description

# Tests 
All unit tests pass? N/A

Builds successfully? Yes 

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
Profile Link would accpet any random characters for now
image
After loading, a list of publications would be populated from the json file, a checkbox attahed to each of those
image
After the import button is clicked, the selected publications will be sent to db and displayed on the page 
image

Error message will pop up when there is error from the scholar api call 
image
* to test this feature, uncomment line 117-120 and comment out line 113-116
image

# Relevant document/ link (if any) 
N/A

# Risk
Low for this PR alone, will be integared to the actual google scholar api.

### CODE REVIEW

**Review by rsun0013 — APPROVED** [review]

File: client/src/components/publications/importedPublication/ImportedPublication.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { InputGroup, Form, Card, Collapse } from 'react-bootstrap';
+import React, { useState } from 'react';
+
+const ImportedPublciation = ({pub, index, setChecked}) => {
```
Comment (vgoh1) [inline client/src/components/publications/importedPublication/ImportedPublication.js:4]: Typo for publication here

File: client/src/reducers/importedPublicationsReducer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { IMPORT_REQUEST, IMPORT_SUCCESS, IMPORT_FAIL, IMPORT_CLEAR_STATE } from '../actions/types'
+
+const initialState = {
+    loading: false,
+    importStatus: null,
+    publications: [],
+    error: null
+}
+
+const importedPublciationReducer = (state=initialState, action) => {
```
Comment (vgoh1) [inline client/src/reducers/importedPublicationsReducer.js:10]: Typo for publication here

File: api/src/controllers/publications.js [hunk 3]
```diff
@@ -150,4 +151,17 @@ async function readAllPublicationsByTeam(req, res) {
     res.status(200).json(foundPublication);
 }
 
-module.exports = {deletePublication, updatePublication, createPublication, readPublication, readAllPublicationsByTeam};
\ No newline at end of file
+async function importPublications(req, res) {
```
Comment (aksh0001) [inline api/src/controllers/publications.js]: nit: documentation required for this controller

**Comment by vgoh1** [comment]
Besides the spelling mistakes + the static analyzer issues, great work! Please fix those up and then I'll hit approve. Thanks for adding the function to do the mass insert in the backend as well.

File: api/src/controllers/publications.js [hunk 4]
```diff
@@ -150,4 +151,17 @@ async function readAllPublicationsByTeam(req, res) {
...
+    const {team_id: _id} = req.params;
+
+    var result = await Team.findById(_id);
```
Comment (aksh0001) [inline api/src/controllers/publications.js:157]: nit: prefer `const` over `var`

File: api/src/controllers/publications.js [hunk 5]
```diff
@@ -150,4 +151,17 @@ async function readAllPublicationsByTeam(req, res) {
...
+    if (result == null) {
+        return res.status(404).send('Error: Team not found.');
+    }
+
```
Comment (aksh0001) [inline api/src/controllers/publications.js:161]: this validation would be a good candidate for middleware as we used previously?

File: client/src/components/publications/Publications.js [hunk 6]
```diff
@@ -5,9 +5,9 @@
 import React, { useEffect, useState } from 'react';
 import { useSelector, useDispatch } from 'react-redux'
 import { getPublicationsByTeamId, sortPublications } from '../../actions/publications'
-import { Button, Modal, InputGroup, FormControl, Dropdown, DropdownButton, Container, Col, Row, Spinner, Alert } from 'react-bootstrap';
+import { Button, Modal, Dropdown, DropdownButton, Container, Col, Row, Spinner, Alert } from 'react-bootstrap';
```
Comment (aksh0001) [inline client/src/components/publications/Publications.js:8]: nit: Static Analyzer warnings about unused imports.

File: client/src/components/publications/form/ImportFailPage.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+import { Alert, Row, Button } from "react-bootstrap";
```
Comment (aksh0001) [inline client/src/components/publications/form/ImportFailPage.js:1]: File header that describes what the page does would be nice.

File: client/src/components/publications/form/ImportForm.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+const ImportForm = ({closeModal}) => {
+    const { importStatus, loading } = useSelector(state => state.importedPublications)
+
+    const displayResult = () => {
+        switch(importStatus){
+            case "SUCCESS":
+                return <ImportSucessPage closeModal={closeModal}/>
+            case "FAIL":
+                return <ImportFailPage closeModal={closeModal}/>
+        }
```
Comment (aksh0001) [inline client/src/components/publications/form/ImportForm.js:17]: minor: we need a default case here (as per the Static Analyzer warnings)

File: client/src/components/publications/form/ImportSuccessPage.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
+import { useSelector, useDispatch } from 'react-redux'
```
Comment (aksh0001) [inline client/src/components/publications/form/ImportSuccessPage.js:1]: same as above: file headers to describe the component.

File: client/src/components/publications/form/ProfileLinkPage.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
+import { Formik } from "formik";
+import * as yup from "yup";
+import { useDispatch } from 'react-redux'
+import { Row, InputGroup, Button, Form, Spinner } from "react-bootstrap";
```
Comment (aksh0001) [inline client/src/components/publications/form/ProfileLinkPage.js:4]: Static Analyzer warnings unused import

File: client/src/components/publications/importedPublication/ImportedPublication.js (hunk 1 above)
Comment (aksh0001) [reply client/src/components/publications/importedPublication/ImportedPublication.js:4]: +1

File: client/src/components/publications/importedPublication/ImportedPublication.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
+import { InputGroup, Form, Card, Collapse } from 'react-bootstrap';
```
Comment (aksh0001) [inline client/src/components/publications/importedPublication/ImportedPublication.js:1]: File headers needed

File: client/src/components/publications/publicationsLayout/LayoutByCategory.js [hunk 12]
```diff
@@ -1,27 +1,27 @@
 import Publication from '../publication/Publication'
```
Comment (aksh0001) [inline client/src/components/publications/publicationsLayout/LayoutByCategory.js:1]: File headers as above.

**Review by aksh0001 — CHANGES_REQUESTED** [review]
Great job on the import UI, it looks really nice!
I think we could do with a bit more documentation before merging though.

**Review by aksh0001 — APPROVED** [review]

**Review by vgoh1 — APPROVED** [review]

**Pull request merged**